frontend/src/components/Layout/SideNav.tsx

```diff
@@ -1,42 +1,120 @@
 
-const SideNav: React.FC<SideNavProps> = ({ openDrawer, closeDrawer, isExpanded }) => {
-  const handleClick = () => {
-    if (isExpanded) {
-      closeDrawer();
-    } else {
-      openDrawer();
+const SideNav: React.FC<SideNavProps> = ({
+  position,
+  toggleDrawer,
+  isExpanded,
+  deleteOnClick,
+  setShowDrawerChatbot,
+  setIsRightExpanded,
+  messages,
+  clearHistoryData,
+}) => {
+  const [isChatModalOpen, setIsChatModalOpen] = useState(false);
+  const [isFullScreen, setIsFullScreen] = useState(false);
+  const { setMessages } = useMessageContext();
+  const date = new Date();
+  useEffect(() => {
+    if (clearHistoryData) {
+      setMessages([
+        {
+          datetime: `${date.toLocaleDateString()} ${date.toLocaleTimeString()}`,
+          id: 2,
+          message:
+            ' Welcome to the Neo4j Knowledge Graph Chat. You can ask questions related to documents which have been completely processed.',
+          user: 'chatbot',
+        },
+      ]);
     }
-  };
+  }, [clearHistoryData]);
 
+  const handleExpandClick = () => {
+    setIsChatModalOpen(true);
+    setIsFullScreen(true);
+    if (setShowDrawerChatbot && setIsRightExpanded) {
+      setShowDrawerChatbot(false);
+      setIsRightExpanded(false);
+    }
+  };
+  const handleShrinkClick = () => {
+    setIsChatModalOpen(false);
+    setIsFullScreen(false);
+    if (setShowDrawerChatbot && setIsRightExpanded) {
+      setShowDrawerChatbot(true);
+      setIsRightExpanded(true);
+    }
+  };
+  const handleClick = () => {
+    toggleDrawer();
+  };
   return (
-    <div
-      style={{
-        height: 'calc(100vh - 60px)',
-        minHeight: '200px',
-        display: 'flex',
-      }}
-    >
-      <SideNavigation iconMenu={true} expanded={false}>
+    <div style={{ height: 'calc(100vh - 58px)', minHeight: '200px', display: 'flex' }}>
+      <SideNavigation iconMenu={true} expanded={false} position={position}>
         <SideNavigation.List>
           <SideNavigation.Item
-            onClick={() => {
-              handleClick();
-            }}
+            onClick={handleClick}
             icon={
               isExpanded ? (
-                <ArrowLeftIconOutline className='n-w-6 n-h-6' />
+                position === 'left' ? (
+                  <ArrowLeftIconOutline />
+                ) : (
+                  <ArrowRightIconOutline />
+                )
+              ) : position === 'left' ? (
+                <ArrowRightIconOutline />
               ) : (
-                <ArrowRightIconOutline className='n-w-6 n-h-6' />
+                <ArrowLeftIconOutline />
               )
             }
-          ></SideNavigation.Item>
+          />
+          {position === 'right' && (
+            <>
+              <SideNavigation.Item icon={<TrashIconOutline className='n-size-full' onClick={deleteOnClick} />} />
+              <SideNavigation.Item icon={<ExpandIcon className='n-size-full' />} onClick={handleExpandClick} />
```

Comment:
don't show this if isExpanded is false
(if chatbot drawer is not expanded, it might be confusing to see a trash or arrow, without knowing those are related to chatbot, hence only show when the chatbot is open)